client/js/components/home.jsx

```diff
@@ -158,20 +165,44 @@ class Home extends React.Component {
   }
 
   offerAssessment(bankId, assessment){
-    this.setState({ isOpen: true,
-                    assessmentClicked: assessment });
+    this.setState({
+      isOpen: true,
+      assessmentClicked: assessment,
+      currentBankId: bankId
+    });
+
     var qBankHost = this.props.settings.qBankHost;
     this.props.offerAssessment(bankId, assessment.id, qBankHost);
+    this.props.getItems(bankId, assessment.id, qBankHost);
+  }
+
+  setNOfM(n) {
+    // If n of m is null, we default it to -1 in the lambda function
+    let nOfM = n;
+    if(n === this.props.items.length) {
+      nOfM = null;
+    }
+
+    this.setState({ nOfM });
```

Comment:
What do you think about adding `nOfM: -1` to the initial state in the constructor?